core/src/components/input/input.ionic.outline.scss

```diff
@@ -0,0 +1,162 @@
+:host(.input-fill-outline.input-label-placement-floating) input {
+  @include margin(6px, 0, 6px, 0);
+}
+
+
+// Input Fill: Outline, Outline Container
+// ----------------------------------------------------------------
+
+/**
+ * Note that while the ionic theme does not make use of the notch,
+ * thus avoiding the need for separate outline container parts at all,
+ * the element's structure has been kept the same to simplify the
+ * component render function and keep selectors consistent between
+ * the ionic and md themes.
+ */
+
+:host(.input-fill-outline) .input-outline-container {
+  @include position(0, 0, 0, 0);
+
+  display: flex;
+
+  position: absolute;
+
+  width: 100%;
+  height: 100%;
+}
+
+:host(.input-fill-outline) .input-outline-start,
+:host(.input-fill-outline) .input-outline-end {
+  pointer-events: none;
+}
+
+/**
+ * Each piece of the container should have a top and bottom
+ * border. This gives the appearance of a unified container
+ * with a border.
+ */
+:host(.input-fill-outline) .input-outline-start,
+:host(.input-fill-outline) .input-outline-notch,
+:host(.input-fill-outline) .input-outline-end {
+  border-top: var(--border-width) var(--border-style) var(--border-color);
+  border-bottom: var(--border-width) var(--border-style) var(--border-color);
+}
+
+/**
+ * Ensures long labels do not cause the notch to flow
+ * out of bounds.
+ */
+:host(.input-fill-outline) .input-outline-notch {
+  max-width: calc(100% - var(--padding-start) - var(--padding-end));
+}
+
+/**
+ * In the md theme, this element ensures that the notch
+ * uses the size of the scaled text so that the border
+ * cut-out is the correct width. However, a notch is not
+ * shown in the ionic theme, so we only need to make sure
+ * the text in this element is not visible or interactive.
+ */
+:host(.input-fill-outline) .notch-spacer {
+  opacity: 0;
+  pointer-events: none;
+}
+
+:host(.input-fill-outline) .input-outline-start {
+  @include border-radius(var(--border-radius), 0px, 0px, var(--border-radius));
+  @include border(null, null, null, var(--border-width) var(--border-style) var(--border-color));
+
+  width: calc(var(--padding-start));
+}
+
+:host(.input-fill-outline) .input-outline-end {
+  @include border(null, var(--border-width) var(--border-style) var(--border-color), null, null);
+  @include border-radius(0px, var(--border-radius), var(--border-radius), 0px);
+
+  /**
+   * The ending outline fragment
+   * should take up the remaining free space.
+   */
+  flex-grow: 1;
+}
+
+
+// Input Fill: Outline, Label Placement: Stacked
+// ----------------------------------------------------------------
+
+// Give room for the label to sit above the outline container.
+:host(.input-fill-outline.input-label-placement-stacked) {
+  // !important is needed to overwrite another !important on the host/base styles.
+  /* stylelint-disable-next-line declaration-no-important */
+  padding-top: 20px !important;
```

Comment:
1. Why are we not using our padding mixin here and disabling stylelint?
2. Should we correct the usage of `!important` in the parent file instead of continuing the practice?

Reply:
1. I didn't realize you could use `!important` with the mixin, and it also isn't required for `padding-top` (or `bottom` for that matter) since the mixin's purpose is to add RTL support. I went ahead and changed it in https://github.com/ionic-team/ionic-framework/pull/29268/commits/2288f1a3116b720be0f9f59976e3530496af8ca4 though.
2. Should we? It seemed out of scope for this PR since the base styles aren't related to the feature and I assume the original `!important` was put there for a reason. I can make a tech debt ticket if you want?

Reply:
I wound up removing this code in https://github.com/ionic-team/ionic-framework/pull/29268/commits/038f2f99cd5ae74b71fd5262b127947555583ff2 anyway, so marking this as resolved. See https://github.com/ionic-team/ionic-framework/pull/29268#discussion_r1552182072 for relevant discussion.